crates/iceberg-catalog/migrations/20250108131004_statistics.sql

```diff
@@ -0,0 +1,117 @@
+CREATE OR REPLACE FUNCTION get_stats_date_default() RETURNS timestamptz AS
+$$
+BEGIN
+    RETURN date_trunc(get_stats_interval_unit(), now()) + get_stats_interval();
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION get_stats_interval_unit() RETURNS text AS
+$$
+BEGIN
+    RETURN 'hour';
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION get_stats_interval() RETURNS interval AS
+$$
+BEGIN
+    RETURN interval '1 ' || get_stats_interval_unit();
+END;
+$$ LANGUAGE plpgsql;
+
+create table warehouse_statistics
+(
+    stats_id         int generated always as identity primary key,
```

Comment:
I think we should use a UUID instead (`uuid_generate_v1mc()`).
Identity can feel weird as they are not consecutive in the domain of interest - the `warehouse_id`.

Do we gain anything from identity?

Reply:
not really, was looking for something that'd retain at least some locality with lack of uuid v7 support in pg, didnt' realize that v1mc is somewhat like a v7, changing it now.